### ISSUE TITLE
TextField: Fails a11y validation :: aria-describedby refer to element not available in the dom

### ISSUE THREAD

**Issue by mdahamiwal** [OfficeDev/office-ui-fabric-react#764]
### Bug Report

- __Package version(s)__: all
- __Browser and OS versions__: all

#### Describe the issue:

TextField uses `aria-describedby` to link description or error message for the input element. For cases where no description or error message is available, aria-describedby refers to an unavailable id since description holder is not rendered. This fails the accessibility validation for the page. https://dequeuniversity.com/rules/axe/2.1/aria-valid-attr-value

I think this is the case with many other components where aria-* attributes refer to Ids that are not part of dom yet or those elements are rendered conditionally.

Step to repro:
1. Install  [Accessibility Developer tools](https://chrome.google.com/webstore/detail/accessibility-developer-t/fpkknkljclfencbdbgkenhalefipecmb) for chrome extension.
2. From chrome developer tools run accessibility Audit.
